File: session06/python/after.cpp
#include <string>
using namespace std;



class Animal{
    public:
        int age;
    Animal(int age): age(age) {}
};

class Person: public Animal {
    public:
    string job;
    Person(int age, string job):Animal(age),job(job){}
};

class Pet: public Animal {
    public:
    Person & owner;
    Pet(int age, Person & owner):
        Animal(age), owner(owner)
    {}
};


int main(int argc, char **argv){
    Person me(37,"Programmer");
    Pet daisy(5,me);
}

File: session06/python/old_fragments/after.py

class Animal(object):
    def __init__(self, age): 
        self.age = age

    def birthday(self): 
        self.age += 1

class Person(Animal):
    def __init__(self, age, job):
        self.job = job
        super(Person, self).__init__(age)


me=Person(37,"Programmer")
me.birthday()
assert(me.age == 38)
